digitalocean/vpcpeering/resource_vpcpeering.go

```diff
@@ -0,0 +1,199 @@
+package vpcpeering
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/digitalocean/godo"
+	"github.com/digitalocean/terraform-provider-digitalocean/digitalocean/config"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceDigitalOceanVPCPeering() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDigitalOceanVPCPeeringCreate,
+		ReadContext:   resourceDigitalOceanVPCPeeringRead,
+		UpdateContext: resourceDigitalOceanVPCPeeringUpdate,
+		DeleteContext: resourceDigitalOceanVPCPeeringDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the VPC Peering",
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The name of the VPC Peering",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"vpc_ids": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The list of VPCs to be peered",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Set: schema.HashString,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the VPC Peering",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time when the VPC was created",
+			},
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Delete: schema.DefaultTimeout(2 * time.Minute),
+		},
+	}
+}
+
+func resourceDigitalOceanVPCPeeringCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+
+	name := d.Get("name").(string)
+	vpcIDs := d.Get("vpc_ids").(*schema.Set).List()
+
+	vpcIDsString := make([]string, len(vpcIDs))
+	for i, v := range vpcIDs {
+		vpcIDsString[i] = v.(string)
+	}
+
+	vpcPeeringRequest := &godo.VPCPeeringCreateRequest{
+		Name:   name,
+		VPCIDs: vpcIDsString,
+	}
+
+	log.Printf("[DEBUG] VPC Peering create request: %#v", vpcPeeringRequest)
+	vpcPeering, _, err := client.VPCs.CreateVPCPeering(context.Background(), vpcPeeringRequest)
+	if err != nil {
+		return diag.Errorf("Error creating VPC Peering: %s", err)
+	}
+
+	d.SetId(vpcPeering.ID)
+
+	log.Printf("[DEBUG] Waiting for VPC Peering (%s) to become active", d.Get("name"))
+	stateConf := &retry.StateChangeConf{
+		Delay:      10 * time.Millisecond,
```

Comment:
I added here an initial delay because if it fetches the state of the peering immediately it will get a not found error and fail the creation. I observed that 10 milliseconds is a good delay time so the peering is added to the db, but if you feel it is too tight we can go for 30 milliseconds. I think 5 seconds is a lot of unnecessary wait time that will make creation take more time than necessary due to state validation